plugins/woocommerce/includes/data-stores/class-wc-product-data-store-cpt.php

```diff
@@ -265,7 +265,7 @@ public function read( &$product ) {
 		$this->read_extra_data( $product );
 		$product->set_object_read( true );
 
-		do_action( 'woocommerce_product_read', $product->get_id() );
+		do_action( 'woocommerce_product_read', $product->get_id(), $product );
```

Comment:
Thanks for proposing this change. This meshes nicely with our current action/filter best practices [(we should prefer passing objects)](https://github.com/woocommerce/woocommerce/blob/9.3.3/docs/extension-development/adding-actions-and-filters.md#prefer-passing-objects-over-ids). 

Can you also add a docblock? Though it was already missing, it is now required for linting to pass. Something like:

```suggestion
		/**
		 * Fires when a product is read into memory.
		 *
		 * @since 3.7.0
		 * @since 9.5.0 Made $product availabe.
		 *
		 * @param int        $product_id The product ID.
		 * @param WC_Product $product    Product instance.
		 */
		do_action( 'woocommerce_product_read', $product->get_id(), $product );
```